setup.py

```diff
@@ -43,7 +43,7 @@
         # as `google-api-core[grpc]`. We thus need to explicitly specify it here.
         # See: https://github.com/googleapis/python-bigquery/issues/83
         "grpcio >= 1.8.2, < 2.0dev",
-        "pyarrow>=0.16.0, < 2.0dev",
+        "pyarrow >= 1.0.0, < 2.0dev; python_version >= '3.5'",
```

Comment:
Python 2 users will be likely stuck with `pyarrow` 0.16.0 (the last release compatible with Python 2), since bqstorage [still supports](https://github.com/googleapis/python-bigquery-storage/blob/master/setup.py#L33) that version.

IIRC that should still be fine, though, as we have a conditional check in the code to only use the `timestamp_as_object` parameter under Python 3 (that parameter is not supported in pyarrow versions compatible with Py2).